### ISSUE TITLE
[JOSS review] MolSolvator minor improvements

### ISSUE THREAD

**Issue by HubLot** [gpantel/MolPainter#20]
This issue is part of the JOSS review process available here: https://github.com/openjournals/joss-reviews/issues/5128

---

Overall, Molsolvator is easy to use with the input config file and produced correct result with the tutorial files provided.
I just have some minor suggestions:

   - **Molsolvator code**: I'm wondering about the use of inner functions inside the code? I believe that they should be *regular* functions so they can be easily tested (inner functions are really difficult to test because they can't be called from outside). Those functions could also have docstrings (as the ones in MolPainter).
   - **Molsolvator crashes** when the input files mentioned in the toml file are not found:
```
Traceback (most recent call last):
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/bin/molsolvator", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MolSolvator/MolSolvator.py", line 221, in main
    solvent_universe = create_solvent_universe(path, solvent_molecule_id, num_solvent)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MolSolvator/MolSolvator.py", line 95, in create_solvent_universe
    solvu = mda.Universe(path)
            ^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/core/universe.py", line 346, in __init__
    topology = _topology_from_file_like(self.filename,
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/core/universe.py", line 118, in _topology_from_file_like
    raise sys.exc_info()[1] from err
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/core/universe.py", line 110, in _topology_from_file_like
    topology = p.parse(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/topology/PDBParser.py", line 207, in parse
    top = self._parseatoms()
          ^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/topology/PDBParser.py", line 244, in _parseatoms
    with util.openany(self.filename) as f:
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/lib/util.py", line 318, in openany
    stream = anyopen(datasource, mode=mode, reset=reset)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/lib/util.py", line 397, in anyopen
    stream = _get_stream(datasource, openfunc, mode=mode)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/lib/util.py", line 439, in _get_stream
    raise sys.exc_info()[1] from err
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/lib/util.py", line 433, in _get_stream
    stream = openfunction(filename, mode=mode)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/lib/picklable_file_io.py", line 478, in bz2_pickle_open
    binary_file = BZ2Picklable(name, bz_mode)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/MDAnalysis/lib/picklable_file_io.py", line 266, in __init__
    super().__init__(name, mode)
  File "/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/bz2.py", line 81, in __init__
    self._fp = _builtin_open(filename, mode)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'tutorial/PDBfiles/NA+.pdb'

``` 

A proper exit mechanism will allow users to quickly understand their mistake.
   - **Warning displayed**: There is a bunch of MDAnalysis warnings when solvating the system. I believe there are harmless. But, there is also 2 numpy warnings that, at this point, I don't know if they are important or not:
```
Inserting 54 molecules of tutorial/PDBfiles/NA+.pdb
/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/numpy/core/fromnumeric.py:3464: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
/ibpc/sablons/santuz/softs/miniconda3/envs/molpainter/lib/python3.11/site-packages/numpy/core/_methods.py:184: RuntimeWarning: invalid value encountered in divide
  ret = um.true_divide(
```

All tests have been conducted under python 3.11, numpy==1.24.2 and MDA==2.4.2

**Comment by gpantel** [comment]
Thank you for the critical comments! We have updated MolSolvator in several ways.

1) All functions are general now and have a doc string.
2) Spurious MDAnalysis warnings related to topology have been suppressed.
3) The NumPy warning was coming from a NaN that was due to using B-factors for determining the center of geometry of solvent molecules. This was not explained in the documentation or tutorial and we feel it's an unnecessary and confusing function so we removed it -- centers of geometry for placing solvents into the lattice are determined normally now.
4) MolSolvator is now a package that is part of MolPainter such that we can have it directly share the version number with MolPainter. The  molsolvator command line options -v and --version will return the MolPainter version now.
